app/logical/sources/strategies/tumblr.rb

```diff
@@ -161,14 +168,22 @@ def dtext_artist_commentary_desc
     # http://media.tumblr.com/tumblr_m24kbxqKAX1rszquso1_1280.jpg
     # => https://media.tumblr.com/tumblr_m24kbxqKAX1rszquso1_1280.jpg
     def find_largest(url, sizes: SIZES)
-      return url unless url =~ OLD_IMAGE
+      if url =~ OLD_IMAGE
+        candidates = sizes.map do |size|
+          "https://media.tumblr.com/#{$~[:dir]}#{$~[:filename]}_#{size}.#{$~[:ext]}"
+        end
 
-      candidates = sizes.map do |size|
-        "https://media.tumblr.com/#{$~[:dir]}#{$~[:filename]}_#{size}.#{$~[:ext]}"
-      end
+        candidates.find do |candidate|
+          http_exists?(candidate)
+        end
+      elsif url =~ %r{/s\d+x\d+/(\w+\.\w+)$}i
```

Comment:
Regexes should use `\z` instead of `$`. See https://stackoverflow.com/a/5979643.